File: RGR/app/src/main/java/com/example/rgr/Comparator/ComparatorGPS.java
package com.example.rgr.Comparator;

import com.example.rgr.Builder.BuilderGPS;

import java.io.Serializable;
import java.lang.reflect.*;


public class ComparatorGPS implements Comparator, Serializable
{
    @Override
    public int compare(Object o1, Object o2)
    {
        return (int) (((BuilderGPS)o1).latitude - ((BuilderGPS)o2).latitude);
    }
}

File: LR_3/src/git/group/Comparator/ComparatorGPS.kt
package git.group.Comparator

import git.group.Builder.BuilderGPS
import java.io.Serializable

class ComparatorGPS : Comparator, Serializable {
    override fun compare(o1: Any?, o2: Any?): Int {
        return ((o1 as BuilderGPS?)!!.latitude - (o2 as BuilderGPS?)!!.latitude).toInt()
    }
}